src/test/unit/services/pathfinder/eight_schools_test.cpp

```diff
@@ -0,0 +1,291 @@
+  Eigen::RowVectorXd prev_sd_vals(20);
+  prev_sd_vals << 1.93964, 4.77042, 0.95799, 0.842812, 0.963455, 0.948548,
+      1.03149, 0.989, 0.920778, 0.888529, 4.6405, 3.63071, 4.25895, 4.45198,
+      3.90755, 4.23075, 4.56257, 4.22915, 4.37932, 2.28608;
+  Eigen::VectorXd diff_vals(20);
+  diff_vals = ((prev_mean_vals.array() + (2.0 * prev_sd_vals.array())).abs()
+               - mean_vals.array().abs())
+                  .matrix()
+                  .transpose();
+  for (Eigen::Index i = 0; i < 18; i++) {
+    EXPECT_GE(diff_vals(i), 0.0);
+  }
+  /*
+  std::cout << "Diffs\n" << diff_vals.format(CommaInitFmt) << "\n";
+  */
+}
+
+TEST_F(ServicesPathfinderEightSchools, single) {
+  constexpr unsigned int seed = 0;
+  constexpr unsigned int chain = 1;
+  constexpr double init_radius = .7;
+  constexpr double num_elbo_draws = 80;
+  constexpr double num_draws = 100;
+  constexpr int history_size = 6;
+  constexpr double init_alpha = 0.001;
+  constexpr double tol_obj = 1e-12;
+  constexpr double tol_rel_obj = 10000;
+  constexpr double tol_grad = 1e-8;
+  constexpr double tol_rel_grad = 10000000;
+  constexpr double tol_param = 1e-8;
+  constexpr int num_iterations = 2000;
+  constexpr bool save_iterations = false;
+  constexpr int num_eval_attempts = 100;
+  constexpr int refresh = 0;
+  std::ostream empty_ostream(nullptr);
+  stan::test::loggy logger(empty_ostream);
+  stan::test::mock_callback callback;
+  int return_code = stan::services::pathfinder::pathfinder_lbfgs_single(
+      model, context, seed, chain, init_radius, history_size, init_alpha,
+      tol_obj, tol_rel_obj, tol_grad, tol_rel_grad, tol_param, num_iterations,
+      save_iterations, refresh, callback, num_elbo_draws, num_draws,
+      num_eval_attempts, logger, init, parameter, diagnostics);
+
+  Eigen::IOFormat CommaInitFmt(Eigen::StreamPrecision, 0, ", ", ", ", "\n", "",
+                               "", "");
+
+  Eigen::MatrixXd param_vals = std::move(parameter.values_);
+
+  // std::cout << "---- Results  -------" << std::endl;
+  /*
+  std::cout << "Values: \n"
+            << param_vals.format(CommaInitFmt) << "\n";
+  */
+  Eigen::VectorXd mean_vals = param_vals.rowwise().mean().eval();
+  //       std::cout << "Mean Values: \n" <<
+  //       mean_vals.transpose().eval().format(CommaInitFmt) << "\n";
+  Eigen::VectorXd sd_vals = (((param_vals.colwise() - mean_vals)
+                                  .array()
+                                  .square()
+                                  .matrix()
+                                  .rowwise()
+                                  .sum()
+                                  .array()
+                              / (param_vals.cols() - 1))
+                                 .sqrt())
+                                .transpose()
+                                .eval();
+
+  Eigen::MatrixXd r_answer = stan::test::eight_schools_r_answer();
+
+  Eigen::MatrixXd r_constrainted_draws_mat(20, 100);
+  {
```

Comment:
Personally I like having the nested environment because it tells the user none of the things in here will be used outside of this scope. Though maybe that means I should just pull it out and have a function